conda/base/context.py

```diff
@@ -822,7 +822,14 @@ def user_agent(self):
         if self.libc_family_version[0]:
             builder.append("%s/%s" % self.libc_family_version)
         if self.experimental_solver.value != "classic":
-            builder.append("solver/%s" % self.experimental_solver.value)
+            from ..core.solve import _get_solver_class
+
+            user_agent_str = "solver/%s" % self.experimental_solver.value
+            Solver = _get_solver_class()
```

Comment:
Thinking about this more, I think this defeats the purpose of the conda.base namespace, where no imports should happen other than conda.common, conda._vendor and conda.base (at least according to the README: https://github.com/conda/conda/tree/master/conda/base)

Can we do this appending in the place where the user-agent string is used instead?